src/visualizer/graphPanel/GraphPanel.cpp

```diff
@@ -0,0 +1,17 @@
+//
+// Created by stefan on 23.07.2023.
+//
+
+#include "GraphPanel.hpp"
+
+namespace age::visualizer {
+GraphPanel::GraphPanel(QWidget* pParent) noexcept : QWidget(pParent) {}
+
+void GraphPanel::mousePressEvent(QMouseEvent* pEvent) {
+  auto mousePos = pEvent->position();
+  cds::UniquePointer<Vertex> vertex = cds::makeUnique<Vertex>((int) mousePos.x(), (int) mousePos.y(), this);
+  vertex->show();
+  _vertexList.pushBack(std::move(vertex));
```

Comment:
How about:
```suggestion
_vertexList.pushBack(cds::makeUnique<Vertex>(mousePos.x(), mousePos.y(), this))->show();
```

Using RVO to implicitly move, and push/emplace_front/back or insert returns ref to emplaced element